File: lab2/Rot.cpp
#include "Rot.h"
#include "FactoryInitializer.h"

void Rot::Execute(Operands& operands, Tokens& tokens, std::string& output, Reader& reader) {
    int operand1 = operands.GetAndPop();
    int operand2 = operands.GetAndPop();
    int operand3 = operands.GetAndPop();

    operands.Push(operand2);
    operands.Push(operand3);
    operands.Push(operand1);

}

namespace {
    FactoryInitializer<Rot> Registration("rot");
}

File: OOP Java/lab1/src/main/java/org/example/commands/Rot.java
package org.example.commands;

import org.example.utility.Operands;

import java.util.StringTokenizer;

public class Rot implements Command{
    @Override
    public void Execute(StringTokenizer tokens, Operands operands) {
        int num1 = operands.GetAndPop();
        int num2 = operands.GetAndPop();
        int num3 = operands.GetAndPop();
        operands.Push(num2);
        operands.Push(num3);
        operands.Push(num1);
    }

    @Override
    public void Check(StringTokenizer tokens, Operands operands) {

    }

}
